server/src/validators/userValidator.js

```diff
@@ -0,0 +1,17 @@
+
+import { check } from 'express-validator/check';
+
+const userValidator = [
+  check('email')
+    .isEmail()
+    .withMessage('must be a valid email'),
+  check('username')
+    .optional()
```

Comment:
username is required in schema, then why optional here??

Reply:
This is because I will use the same validator for the login API, and username will be optional there. I should probably make this a function that take a parameter denoting if username is optional or not. Thoughts?

Reply:
Yeah, we only want username to be option when logging in and not when signing up. So, that would be a good idea. 

Or we catch error thrown by mongodb and send response to client accordingly. But first approach seems better to me.

Reply:
Okay, I will do that in the login PR.